### PR TITLE
Fix flip transform on drop targets and editing

### LINKED ISSUE #1111: Drop Target not Following Flip
## Bug Description

When you flip an element, the drop target does not follow. 

**Note:** when #1878 is merged, there's an additional problem on droptarget hover, as hovering a non-flipped element now resets the flip of the drop target which results in displaying both a flipped and a non-flipped drop target (only visible with non-regular blobs).

## Expected Behaviour

Drop target should follow modifications to the element. 

## Steps to Reproduce

1. Flip a shape
2. Attempt to mask the shape with an image and see drop target

## Screenshots

flipped drop target

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by wassgha**
Closes #1111

### Changes
- Fixes an issue that caused drop targets and outlines to appear flipped when a transformation is applied
- Simplifies and fixes flipping on crop/edit

### CODE REVIEW

File: assets/src/edit-story/masks/frame.js [hunk 1]
```diff
@@ -127,9 +128,22 @@ WithDropTarget.propTypes = {
...
+  children,
+  ...rest
+}) {
   const [hover, setHover] = useState(false);
-  const { isBackground } = element;
+  const { flip, isBackground } = element;
+
+  const transformFlip = getTransformFlip(flip);
+  if (transformFlip) {
+    style.transform = style.transform
```
Comment (dvoytenko) [inline assets/src/edit-story/masks/frame.js:143]: Please clone (not mutate) the prop.

**Review by dvoytenko — APPROVED** [review]
Approved with one nit

**Comment by barklund** [comment]
I found a really weird bug with a flipped mask and edit mode - I don't know if it's directly related to this one though or should be filed as a new one.

When entering edit mode for a flipped heart with an image inside, the image moved inside the mask. When I start dragging the image inside the mask, things start to get reaaaally funky!

flipped-mask-bug

**Comment by wassgha** [comment]
Yeah I saw that, working on a fix in this PR

**Review by miina — CHANGES_REQUESTED** [review]
Please flip a video or image, enter edit mode and try panning - the panned image is not flipped.

`applyFlip` logic was there for a reason :)

**Comment by miina** [comment]
@wassgha Just saw that Morten already mentioned the same bug, sorry for the repetition.

**Comment by swissspidy** [comment]
@wassgha Hey Wassim, what's the status for this one?

**Pull request closed without merging**